### ISSUE TITLE
`position_ids` related PPO bug

### ISSUE THREAD

**Issue by tianhao-nexusflow** [OpenLLMAI/OpenRLHF#217]
I'm working to reproduce results and align them with `trlx`. After careful code review, I've identified this issue:

Concern: [link](https://github.com/OpenLLMAI/OpenRLHF/blob/ecfb6e8704a4e7d0c82f1f1134a96aa3b2b62d2d/openrlhf/trainer/ppo_utils/experience_maker.py#L124), Precomputing log probabilities using only `token_ids` and `attention_mask` may not fully account for the model's use of positional embeddings. As for llama model, the same `token_ids` with or without left padding would have different result even it explicitly pass in the `attention_mask`. A common strategy to correct this is to explicitly pass in the `position_ids` as in [link](https://github.com/CarperAI/trlx/blob/3340c2f3a56d1d14fdd5f13ad575121fa26b6d92/trlx/trainer/accelerate_ppo_trainer.py#L180)

Potential Impact: The same padding related issue could also impact the generate quality, while I need more time to carefully look into the code. Will this also impact the critic and reward modules?

**Comment by hijkzzz** [comment]
This issue is very interesting and I will confirm it as soon as possible. If so, the bug is caused by huggingface and we will fix it

**Comment by hijkzzz** [comment]
fixed
